### PR TITLE
feat: query OS Features for classified roads that match an USRN

### PR DESCRIPTION
**Pull request by jessicamcinchak**
**Changes:**
- switch from using a buffered site boundary (#1467) to matching on the USRN associated with this site's UPRN to fetch road classifications associated with an address (I've intentionally opened as a separate PR here for side-by-side testing if needed)
  - this means we'll only fetch classified roads in cases where an applicant has picked an OS address from the autocomplete dropdown, we won't be able to fetch classified roads info when plotting a new, non-UPRN address on the map (they'll see question instead)
  - this clears up previous testing concerns around a buffer picking up classified roads that the property does not "front", as the UPRN will only match a single USRN 
  - see comment threads in this Trello ticket for full context https://trello.com/c/EvEuV0lm/2105-implement-pull-in-road-classifications

**Matching on USRN means a few things:**
- even though UPRN <> USRN is a one to one match, there can be many road segments per USRN. For example, 1 segment intersects with buffered boundary, but 32 segments match USRN for 409 Amersham Rd address in Bucks - can we safely limit 1 and assume the same classification applies to every segment? docs & examples I've looked at seem like "yes" 
- OS Features `typeNames=Highways_RoadLink` which is the source that returns classification attributes does _not_ return the USRN as it's own field, rather it's referenced in `<osfeatures:FormsPartOf>` - which means our OGC XML filter can't use `PropertyIsEqualTo` but needs to use `PropertyIsLike` which is noticibly slow

timing profile for http://localhost:7002/roads?usrn=45500010 (409 Amersham Rd HP15 7JG on a classified road): 
- return all matching results (32 features): 31 seconds
- return first matching result only: 5.5 seconds

timing profile for http://localhost:7002/roads?usrn=22500531 (47 Cobourg Rd SE5 0HU not on a classified road): 13 seconds

**Next steps:** 
- get this back to PO testing using the USRN attribute filter and `count: 1` methodology
- IF feedback says too slow or can't safely limit one, then I think the next route to explore would be: 
  - split into two calls: first call the [OS Linked Identifiers API](https://osdatahub.os.uk/docs/linkedIdentifiers/overview) to go from USRN <> some Highways ID that has its own explicit field. Then call OS Features to using `PropertyIsEqualTo` filter with that ID. Still going to return many segments I suspect, but should be more efficient/faster than `PropertyIsLike` ? Then, we'd have to decide how to format/return/filter 30+ feature records into a meaningful constraint response.

### CODE REVIEW

**Comment by jessicamcinchak** [comment]
[[Implement] Pull in road classifications](https://trello.com/c/EvEuV0lm/2105-implement-pull-in-road-classifications)

**Review by DafyddLlyr — APPROVED** [review]
Super helpful PR description thanks - outline of next steps is also very useful.

Your suspicion of latency coming from use of `PropertyIsLike` vs `PropertyIsEqual` makes sense to me 👍 

On speed of requests - we could display query results as we get them for the varying calls, giving the user some more feedback that something is happening. Not sure if this is better UX though tbh!

File: api.planx.uk/server.ts [hunk 1]
```diff
@@ -471,6 +472,8 @@ app.get("/gis", (_req, res, next) => {
 
 app.get("/gis/:localAuthority", locationSearch);
 
+app.get("/roads", classifiedRoadsSearch);
```
[the remark this answers is not in the document]
Comment (DafyddLlyr) [reply api.planx.uk/server.ts:475]: I think I have actually asked a similar question before - apologies if you've already given an answer to this one...! 😅

[the remark this answers is not in the document]
Comment (jessicamcinchak) [reply api.planx.uk/server.ts:475]: `/gis/roads` will return 400 and throw error "roads is not a supported local authority" independent of where it's positioned in server.ts unfortunately ! 

my ideal future here is that we simply have `/gis/roads` & `/gis/constraints` and neither require council-specific params - but given the Digital Land onboarding process and scale of available data, I'm not sure we'll be able to move away from the `/gis/:localAuthority` param anytime soon as a filtering mechanism

File: api.planx.uk/gis/classifiedRoads.js
[the remark this answers is not in the document]
Comment (jessicamcinchak) [reply api.planx.uk/gis/classifiedRoads.js:1]: yeah, this is very fair to point out - `/gis/classifiedRoads` is now TS - all types are in-file only for now until I convert the whole module and have a better structure for sharing relevant types between roads <> constraints

**Pull request merged**